dask/bytes/tests/test_s3.py

```diff
@@ -0,0 +1,217 @@
+from __future__ import print_function, division, absolute_import
+
+from contextlib import contextmanager
+
+import pytest
+pytest.importorskip('s3fs')
+
+import boto3
+import moto
+from toolz import concat, valmap, partial
+from s3fs import S3FileSystem
+
+from dask import compute, get
+from dask.bytes.s3 import read_bytes, open_files, getsize
+
+
+compute = partial(compute, get=get)
+
+
+test_bucket_name = 'test'
+files = {'test/accounts.1.json':  (b'{"amount": 100, "name": "Alice"}\n'
+                                   b'{"amount": 200, "name": "Bob"}\n'
+                                   b'{"amount": 300, "name": "Charlie"}\n'
+                                   b'{"amount": 400, "name": "Dennis"}\n'),
+         'test/accounts.2.json':  (b'{"amount": 500, "name": "Alice"}\n'
+                                   b'{"amount": 600, "name": "Bob"}\n'
+                                   b'{"amount": 700, "name": "Charlie"}\n'
+                                   b'{"amount": 800, "name": "Dennis"}\n')}
+
+
+@pytest.yield_fixture
+def s3():
+    # writable local S3 system
+    with moto.mock_s3() as m:
+        client = boto3.client('s3')
+        client.create_bucket(Bucket=test_bucket_name, ACL='public-read-write')
+        for f, data in files.items():
+            client.put_object(Bucket=test_bucket_name, Key=f, Body=data)
+        yield S3FileSystem(anon=True)
+
+
+@contextmanager
+def s3_context(bucket, files):
+    m = moto.mock_s3()
+    m.start()
+    client = boto3.client('s3')
+    client.create_bucket(Bucket=bucket, ACL='public-read-write')
+    for f, data in files.items():
+        client.put_object(Bucket=bucket, Key=f, Body=data)
+
+    yield S3FileSystem(anon=True)
+
+    for f, data in files.items():
+        try:
+            client.delete_object(Bucket=bucket, Key=f, Body=data)
+        except:
+            pass
+    m.stop()
+
+
+def test_read_bytes(s3):
+    sample, values = read_bytes(test_bucket_name+'/test/accounts.*', s3=s3)
+    assert isinstance(sample, bytes)
+    assert sample[:5] == files[sorted(files)[0]][:5]
+
+    assert isinstance(values, (list, tuple))
+    assert isinstance(values[0], (list, tuple))
+    assert hasattr(values[0][0], 'dask')
+
+    assert sum(map(len, values)) >= len(files)
+    results = compute(*concat(values))
+    assert set(results) == set(files.values())
+
+
+def test_read_bytes_blocksize_none(s3):
+    _, values = read_bytes(test_bucket_name+'/test/accounts.*', blocksize=None,
+            s3=s3)
+    assert sum(map(len, values)) == len(files)
+
+
+@pytest.mark.slow
+def test_read_bytes_blocksize_on_large_data():
+    _, L = read_bytes('dask-data/nyc-taxi/2015/yellow_tripdata_2015-01.csv',
```

Comment:
Must leave a note to safeguard this data. Perhaps should also be copied into the test bucket?